plugins/org.jkiss.dbeaver.ui.editors.connection/src/org/jkiss/dbeaver/ui/actions/datasource/DataSourceHandler.java

```diff
@@ -89,6 +94,24 @@ public static void connectToDataSource(
                 return;
             }
 
+            if (DBWorkbench.getPlatform().getApplication().isEarlyAccessProgramActive()) {
```

Comment:
The idea was to limit total amount of connections, not just active ones.
We must check it in new connection create handler.
We also must check that only one project is loaded and restrict creation of more than one project.

Also we cannot use getAllDataSources function (it triggers all projects loading).